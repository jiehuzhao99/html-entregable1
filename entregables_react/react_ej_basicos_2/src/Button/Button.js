const Button = ({ buttonFunction }) => {

    return ( 
        <div>
            <button onClick={buttonFunction}>Click me</button>
        </div>
     );
}
 
export default Button;