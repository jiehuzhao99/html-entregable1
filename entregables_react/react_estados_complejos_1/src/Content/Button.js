import { useLanguageContext } from "../LanguageProvider";

const Button = ({ }) => {
    // const [language, setLanguage] = useState(language);
    const { language, setLanguage } = useLanguageContext();

    const handleChangeLanguage = () => {
        setLanguage(language === 'en'? 'es': 'en');
    }

    return ( 
        <div className="button" style={{ display: 'flex', justifyContent: 'center'}}>
            <p style={{ marginRight: '20px' }}>Button: </p>
            <button style={{ marginRight: '20px' }} onClick={handleChangeLanguage}>Change Language</button>
            <p>{language === 'en'? 'Hello': 'Hola'}</p>
        </div>
     );
}
 
export default Button;