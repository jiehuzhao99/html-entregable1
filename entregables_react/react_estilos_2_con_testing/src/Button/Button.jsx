import styled from "styled-components";

const Button = styled.button`
    background: #fff;
    border: 1px solid #000;
    border-radius: 3px;
    color: #000;
    cursor: pointer;
    font-size: 1rem;
    font-weight: 600;
    padding: 0.5rem 1rem;
    transition: all 0.2s ease-in-out;
`;

export const getButtonStyles = (type) => {
    switch(type) {
        case 'success':
            return styled(Button)`
            background-color: green;
            color: white;
          `;
        case 'error':
          return styled(Button)`
            background-color: red;
            color: white;
          `;
        case 'warning':
          return styled(Button)`
            background-color: yellow;
            color: black;
          `;
        case 'info':
          return styled(Button)`
            background-color: blue;
            color: white;
          `;
        default:
          return Button;
      }
}