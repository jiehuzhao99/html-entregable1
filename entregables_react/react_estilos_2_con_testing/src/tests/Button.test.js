import React from 'react';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
// import { toMatchSnapshot } from "jest-snapshot";


import { getButtonStyles } from '../Button/Button';


test('Button styles to match snapshot', () => {
    const DefaultButton = getButtonStyles();
    const SuccessButton = getButtonStyles('success');
    const ErrorButton = getButtonStyles('error');
    const WarningButton = getButtonStyles('warning');
    const InfoButton = getButtonStyles('info');

    const defaultButtonTree = renderer.create(<DefaultButton>Default</DefaultButton>).toJSON();
    const successButtonTree = renderer.create(<SuccessButton>Success</SuccessButton>).toJSON();
    const errorButtonTree = renderer.create(<ErrorButton>Error</ErrorButton>).toJSON();
    const warningButtonTree = renderer.create(<WarningButton>Warning</WarningButton>).toJSON();
    const infoButtonTree = renderer.create(<InfoButton>Info</InfoButton>).toJSON();

    expect(defaultButtonTree).toMatchSnapshot();
    expect(successButtonTree).toMatchSnapshot();
    expect(errorButtonTree).toMatchSnapshot();
    expect(warningButtonTree).toMatchSnapshot();
    expect(infoButtonTree).toMatchSnapshot();
})