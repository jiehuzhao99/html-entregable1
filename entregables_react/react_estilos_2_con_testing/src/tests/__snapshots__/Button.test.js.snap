// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styles to match snapshot 1`] = `
<button
  className="sc-bgqQcB jqwEdI"
>
  Default
</button>
`;

exports[`Button styles to match snapshot 2`] = `
<button
  className="sc-bgqQcB sc-gTRrQi jqwEdI hObsJv"
>
  Success
</button>
`;

exports[`Button styles to match snapshot 3`] = `
<button
  className="sc-bgqQcB sc-ewnqHT jqwEdI ebpYmZ"
>
  Error
</button>
`;

exports[`Button styles to match snapshot 4`] = `
<button
  className="sc-bgqQcB sc-fFGjHI jqwEdI iimDYr"
>
  Warning
</button>
`;

exports[`Button styles to match snapshot 5`] = `
<button
  className="sc-bgqQcB sc-dicizt jqwEdI brDaxQ"
>
  Info
</button>
`;
